### PR TITLE
Move Views v2 Customizer settings into dedicated class

### PR DESCRIPTION
**Pull request by lucatume**
Issue: https://moderntribe.atlassian.net/browse/TEC-3686 

This PR refactors the existing Customizer code to move the Views v2 specific modifications to Views v2 dedicated Customizer handlers.

### CODE REVIEW

File: src/Tribe/Views/V2/Customizer.php [hunk 1]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+			$accent_color_active         = 'rgba(' . $accent_css_rgb . ',0.9)';
+			$accent_color_background     = 'rgba(' . $accent_css_rgb . ',0.07)';
+			$accent_color_multiday       = 'rgba(' . $accent_css_rgb . ',0.24)';
+			$accent_color_multiday_hover = 'rgba(' . $accent_css_rgb . ',0.34)';
+			$color_background            = '#ffffff';
+
+			// overrides for common base/full/forms/_toggles.pcss.
+			$css_template .= '
+				.tribe-common .tribe-common-form-control-toggle__input:checked {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:64]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: f09ff371275408f4d5291cc1a342a701</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 2]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+			$css_template .= '
+				.tribe-common .tribe-common-cta--alt,
+				.tribe-common .tribe-common-cta--alt:active,
+				.tribe-common .tribe-common-cta--alt:hover,
+				.tribe-common .tribe-common-cta--alt:focus,
+				.tribe-common .tribe-common-cta--thin-alt,
+				.tribe-common .tribe-common-cta--thin-alt:active,
+				.tribe-common .tribe-common-cta--thin-alt:focus,
+				.tribe-common .tribe-common-cta--thin-alt:hover {
+					border-bottom-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:78]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 774838ea0b7f9757fc8f5a678e1b63e4</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 3]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				.tribe-common .tribe-common-cta--alt:hover,
+				.tribe-common .tribe-common-cta--alt:focus,
+				.tribe-common .tribe-common-cta--thin-alt:active,
+				.tribe-common .tribe-common-cta--thin-alt:hover,
+				.tribe-common .tribe-common-cta--thin-alt:focus,
+				.tribe-theme-twentyseventeen .tribe-common .tribe-common-cta--alt:hover,
+				.tribe-theme-twentyseventeen .tribe-common .tribe-common-cta--alt:focus,
+				.tribe-theme-twentyseventeen .tribe-common .tribe-common-cta--thin-alt:hover,
+				.tribe-theme-twentyseventeen .tribe-common .tribe-common-cta--thin-alt:focus {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:93]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 79cff4b8baf96124d67c45566dac7e3c</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 4]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// Overrides for common components/full/buttons/_border.pcss.
+			$css_template .= '
+				.tribe-common .tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt),
+				.tribe-common a.tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt) {
+					border-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:101]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: e914ce2be0dcc0f544204b2f63b369ba</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 5]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:102]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 1b086ff5f86a86fdd62a5a97e39e530c</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 6]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-common .tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt):focus,
+				.tribe-common .tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt):hover,
+				.tribe-common a.tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt):focus,
+				.tribe-common a.tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt):hover {
+					color: ' . $color_background . ';
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:112]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 4d8ce5bd9e7a1b2241badfc430d4a66b</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 7]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-theme-twentyseventeen .tribe-common .tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt):focus,
+				.tribe-theme-twentyseventeen .tribe-common .tribe-common-c-btn-border:not(.tribe-common-c-btn-border--secondary):not(.tribe-common-c-btn-border--alt):hover,
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:119]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 4e68563db5d5e1dcc34c5ebc91552930</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 8]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for common components/full/buttons/_solid.pcss.
+			$css_template .= '
+				.tribe-common .tribe-common-c-btn,
+				.tribe-common a.tribe-common-c-btn {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:127]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 343420770e891ae3bdd10c9376ed0efd</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 9]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				.tribe-common .tribe-common-c-btn:disabled,
+				.tribe-common a.tribe-common-c-btn:disabled {
+					background-color: ' . $accent_color_background . ';
+				}
+			';
+
+			// Override svg icons color.
+			$css_template .= '
+				.tribe-common .tribe-common-c-svgicon {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:157]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: ceebe9fc70387dd82c58d6a724324cb0</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 10]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-common .tribe-events-virtual-virtual-event__icon-svg {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:163]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 3af8c6e291e719455d92f2893cd59097</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 11]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-theme-twentytwenty .tribe-common .tribe-common-c-btn {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:169]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: c6d4bed79c86433d0a461dce52c1eb5a</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 12]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				.tribe-events .datepicker .day.active:hover,
+				.tribe-events .datepicker .day.active:focus,
+				.tribe-events .datepicker .day.active.focused,
+				.tribe-events .datepicker .month.active:hover,
+				.tribe-events .datepicker .month.active:focus,
+				.tribe-events .datepicker .month.active.focused,
+				.tribe-events .datepicker .year.active:hover,
+				.tribe-events .datepicker .year.active:focus,
+				.tribe-events .datepicker .year.active.focused {
+					background: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:213]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: ff93adacdc7f927f53e2da191446bd42</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 13]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec components/full/_events-bar.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-c-events-bar__search-button:before {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:220]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: fc8bcd1b5a0cda3ab6d5c641288c48eb</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 14]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec components/full/_ical-link.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-c-ical__link {
+					border-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:227]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: e49b65e8f464402db0fea1362c3f7fb0</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 15]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:228]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: c0d97bf8f2c577314474a9cc018e2fb3</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 16]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-events .tribe-events-c-ical__link:hover,
+				.tribe-events .tribe-events-c-ical__link:focus,
+				.tribe-events .tribe-events-c-ical__link:active {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:236]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 27a5b9d29bc1c32357df864dd10a716a</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 17]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec components/full/_view-selector.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-c-view-selector__button:before {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:243]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: a2c5ef863c3ef747d82dc7a63b3604f5</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 18]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec views/full/list/_event.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-list__event-row--featured .tribe-events-calendar-list__event-date-tag-datetime:after {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:250]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 6126b31be5c6cca43232880e05647549</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 19]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-common--breakpoint-medium.tribe-events .tribe-events-calendar-list__event-datetime-featured-text {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:256]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 368f67766efb8b9303ca1a1f539a4c27</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 20]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec views/full/month/_calendar-event.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-month__calendar-event--featured:before {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:263]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 7737a10f5d5ad87402cfd63b6a1d866a</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 21]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec views/full/month/_day.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-month__day--current .tribe-events-calendar-month__day-date,
+				.tribe-events .tribe-events-calendar-month__day--current .tribe-events-calendar-month__day-date-link {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:271]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 477d931f6abfd02111f589071ec98125</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 22]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				.tribe-events .tribe-events-calendar-month__day--current .tribe-events-calendar-month__day-date-link:active {
+					color: ' . $accent_color_active . ';
+				}
+			';
+
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-month__day-cell--selected,
+				.tribe-events .tribe-events-calendar-month__day-cell--selected:hover,
+				.tribe-events .tribe-events-calendar-month__day-cell--selected:focus {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:292]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 38bf06eb2a0fd83877282eb58d94cf73</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 23]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-month__day-cell--selected .tribe-events-calendar-month__day-date {
+					color: ' . $color_background . ';
+				}
+			';
+
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-month__mobile-events-icon--event {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:304]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 5f55076473ba6c9dd86a021bbfeb92fc</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 24]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+			$css_template .= '
+				.tribe-theme-twentyseventeen .tribe-events .tribe-events-calendar-month__day--current .tribe-events-calendar-month__day-date-link:active {
+					color: ' . $accent_color_active . ';
+				}
+			';
+
+			$css_template .= '
+				.tribe-theme-twentyseventeen .tribe-events .tribe-events-calendar-month__day-cell--selected:hover,
+				.tribe-theme-twentyseventeen .tribe-events .tribe-events-calendar-month__day-cell--selected:focus {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:324]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 4e9d1866af166f76cc113e62e21dde4e</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 25]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-theme-twentytwenty .tribe-events .tribe-events-calendar-month__day-cell--selected {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:330]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 7118566ff2082ea497fde07f344492f1</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 26]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			// overrides for tec views/full/month/_mobile-events.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-month-mobile-events__mobile-event-datetime-featured-text {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:337]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 514a87993b48915afb2f8d0fe58a7b07</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 27]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				.tribe-events .tribe-events-calendar-month__multiday-event-bar-inner--hover,
+				.tribe-events .tribe-events-calendar-month__multiday-event-bar-inner--focus {
+					background-color: ' . $accent_color_multiday_hover . ';
+				}
+			';
+
+			// overrides for tec views/full/day/_event.pcss.
+			$css_template .= '
+				.tribe-events .tribe-events-calendar-day__event--featured:after {
+					background-color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:358]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 5ad152a227aa2b1da6206f55b3f03d21</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 28]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				}
+			';
+
+			$css_template .= '
+				.tribe-common--breakpoint-medium.tribe-events .tribe-events-calendar-day__event-datetime-featured-text {
+					color: <%= global_elements.accent_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:364]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: fa1caded31a9f7bbaae5a01be7e93ce2</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 29]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+				.tribe-events-event-url a,
+				.tribe-venue-url a,
+				.tribe-organizer-url a,
+
+				.tribe-events-pro .tribe-events-pro-organizer__meta-website a,
+				.tribe-block__organizer__website a,
+				.tribe-events-pro .tribe-events-pro-venue__meta-website a,
+				.tribe-block__venue__website a,
+				.tribe_events p a {
+					color: <%= global_elements.link_color %>;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:381]: :warning: Found Underscore.js unescaped output notation: "<%=".

<sub>hash: 536c8fb0058ae1a7d6c47ee88c96303f</sub>

File: src/Tribe/Views/V2/Customizer.php [hunk 30]
```diff
@@ -23,12 +23,381 @@ class Customizer {
...
+	 *
+	 * @since TBD
+	 *
+	 * @param string                     $css_template The CSS template, as produced by the Global Elements.
+	 * @param Tribe__Customizer__Section $section      The Single Event section.
+	 * @param Tribe__Customizer          $customizer   The current Customizer instance.
+	 *
+	 * @return string The filtered CSS template.
+	 */
+	public function filter_single_event_css_template(  $css_template, $section, $customizer) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Customizer.php:400]: * :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: Expected 1 spaces after opening parenthesis; 2 found
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space before closing parenthesis is prohibited

<sub>hash: 5a390ed3841dded39878e0f7a25aeecb</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 31]
```diff
@@ -864,4 +867,42 @@ public function filter_customizer_sections( $sections, $customizer ) {
...
+	 *
+	 * @since TBD
+	 *
+	 * @param string                     $css_template The CSS template, as produced by the Global Elements.
+	 * @param Tribe__Customizer__Section $section      The Global Elements section.
+	 * @param Tribe__Customizer          $customizer   The current Customizer instance.
+	 *
+	 * @return string The filtered CSS template.
+	 */
+	public function filter_global_elements_css_template($css_template, $section, $customizer  ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php:882]: * :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: Expected exactly one space before closing parenthesis; "  " found.
* :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space after the first parenthesis

<sub>hash: 488062661f66e37e0fd7dc94b3d9a276</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 9 error(s)
:warning: 29 warning(s)

**Review by alliuca — APPROVED** [review]

**Pull request merged**